discussion_grading/discussion_grading.py

```diff
@@ -2,33 +2,147 @@
 
 from __future__ import annotations
 
+import logging
+from typing import Optional
+
 import pkg_resources
 from django.utils import translation
+from requests.exceptions import ConnectionError as RequestsConnectionError
 from web_fragments.fragment import Fragment
+from xblock.completable import CompletableXBlockMixin
 from xblock.core import XBlock
-from xblock.fields import Scope, String
+from xblock.fields import Float, Integer, Scope, String
 from xblock.utils.resources import ResourceLoader
 from xblock.utils.studio_editable import StudioEditableXBlockMixin
 
-from discussion_grading.utils import _
+from discussion_grading.edxapp_wrapper.comments import get_course_user_stats
+from discussion_grading.edxapp_wrapper.submissions import create_submission, get_score, set_score
+from discussion_grading.enums import DiscussionGradingMethod
+from discussion_grading.utils import _, get_anonymous_user_id, get_username
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
 
+ITEM_TYPE = "discussion_grading"
+MAX_SCORE = 1
+MIN_SCORE = 0
 
-class XBlockDiscussionGrading(StudioEditableXBlockMixin, XBlock):
+
+@XBlock.needs("i18n")
+@XBlock.needs("user")
+class XBlockDiscussionGrading(StudioEditableXBlockMixin, CompletableXBlockMixin, XBlock):
     """
     DiscussionGrading XBlock provides a way to grade discussions in Open edX.
     """
 
     CATEGORY = "discussion_grading"
 
+    has_score = True
+    icon_class = "problem"
+
     display_name = String(
         display_name=_("Display Name"),
         help=_("The display name for this component."),
-        scope=Scope.settings,
         default=_("Discussion Grading"),
+        scope=Scope.settings,
+    )
+
+    grading_method = String(
+        display_name=_("Grading Method"),
+        help=_(
+            "Defines the grading method to be used. When set to 'Minimum "
+            "Participations', the learner will obtain the maximum score if the "
+            "number of participations is greater than or equal to the number of "
+            "participations required to pass, 0 otherwise. When set to 'Average "
+            "Participations', the learner will obtain a score equal to the number "
+            "of participations divided by the number of participations required "
+            "to pass."
```

Comment:
Let's put a pin here: we should add how it rounds (to the closest integer, ...) depending on what the product owner says